Lib/test/test_types.py

```diff
@@ -665,6 +665,11 @@ def test_or_type_operator_with_TypeVar(self):
         assert TV | str == typing.Union[TV, str]
         assert str | TV == typing.Union[str, TV]
 
+    def test_union_parameter_chaining(self):
+        T = typing.TypeVar("T")
+
+        assert (float | list[T])[int] == float | list[int]
```

Comment:
I recommend using the unittest assert for better error messages:
```suggestion
        self.assertEqual((float | list[T])[int], float | list[int])
```